tutorials/performance/pipeline_compilations.rst

```diff
@@ -0,0 +1,77 @@
+
+If your game makes use of these features, **make sure to have an scene that uses them as early as possible** before loading the majority of the assets. This scene can be very simple and will do the job as long as it uses the features the game plans to use. It can even be rendered off-screen for at least one frame if necessary. You should also keep in mind that changing any of these features during gameplay will result in immediate stutters. Make sure to only change these features from configuration screens if necessary and insert loading screens and messages when the changes are applied.
+
+- **MSAA Level**: Enabled when the level of 3D MSAA is changed on the project settings. Unfortunately, different MSAA levels being used on different viewports will lead to stutters as the engine only keeps track of one level at a time to perform pre-compilation.
+- **Reflection Probes**: Enabled when a ReflectionProbe node is placed on the scene.
+- **Separate Specular**: Enabled when using effects like sub-surface scattering or a compositor effect that relies on sampling the specularity directly off the screen.
+- **Motion Vectors**: Enabled when using effects such as TAA, FSR2 or a compositor effect that requires motion vectors (such as motion blur).
+- **Normal and Roughness**: Enabled when using an effect or shader that relies on sampling the normals directly off the screen.
+- **Lightmaps**: Enabled when a LightmapGI node is placed on the scene and a node uses a baked lightmap.
+- **VoxelGI**: Enabled when a VoxelGI node is placed on the scene.
+- **SDFGI**: Enabled when the WorldEnvironment enables SDFGI.
+- **Multiview**: Enabled for XR projects.
+- **16/32-bit Shadows**: Enabled when the configuration of the depth of shadowmaps is changed on the project settings.
+- **Omni Shadow Dual Paraboloid**: Enabled when an omni light casts shadows and uses the dual paraboloid mode.
+- **Omni Shadow Cubemap**: Enabled when an omni light casts shadows and uses the cubemap mode.
+
+If you witness stutters during gameplay and the monitors report a sudden increase in compilations during the **Surface** step, it is very likely a feature was not enabled ahead of time. Making use of this effect in the initial scene that your game loads will likely mitigate the issue.
+
+Pipeline Pre-Compilation Instancing
+-----------------------------------
+
+One common source of stutters in games is the fact that some effects are only instanced on the scene because of interactions that only happen during gameplay. This is often the case with particle effects, which are only added to the scene through scripts when the player does an action. Even if the scene is preloaded, the engine might be unable to pre-compile the pipelines until the effect is added to the scene at least once. Luckily, it's possible for **Godot 4.4** to pre-compile these pipelines as long as the scene is instantiated at least once on the scene, even if it's completely invisible or outside of the camera's view.
```

Comment:
```suggestion
One common source of stutters in games is the fact that some effects are only instanced on the scene because of interactions that only happen during gameplay. For example, if you have a particle effect that is only added to the scene through a script when a player does an action. Even if the scene is preloaded, the engine might be unable to pre-compile the pipelines until the effect is added to the scene at least once. Luckily, it's possible for **Godot 4.4** to pre-compile these pipelines as long as the scene is instantiated at least once on the scene, even if it's completely invisible or outside of the camera's view.
```